packages/core/rendering/core-rendering-server-internal/src/rendering_service.tsx

```diff
@@ -101,14 +103,36 @@ export class RenderingService {
     const buildNum = env.packageInfo.buildNum;
     const basePath = http.basePath.get(request);
     const { serverBasePath, publicBaseUrl } = http.basePath;
+
+    let settingsUserValues: Record<string, UserProvidedValues> = {};
+    let globalSettingsUserValues: Record<string, UserProvidedValues> = {};
+    let userSettingsUserValues: Record<string, string> = {};
+
+    if (!isAnonymousPage) {
+      const userValues = await Promise.all([
+        uiSettings.client?.getUserProvided(),
+        uiSettings.globalClient?.getUserProvided(),
+        uiSettings.userClient?.getUserProfileSettings(request),
```

Comment:
The other type of clients are already scoped to a request. the `user` client shouldn't be different. Having to pass the request here is a design flaw ihmo.

Reply:
@majagrubic @pgayvallet How can I best get the request to the `UserProfileSettingsClient`?

@majagrubic had [similar concerns](https://github.com/elastic/kibana/pull/151507#discussion_r1126271094) when I was passing the `request` from `CoreUiSettingsRouteHandlerContext` > `UiSettingServiceStart.userAsScopedToClient()` into the constructor of the `UiSettingsUserClient`

Reply:
Your plugin `security` should handle that. Plugin should initialize the `UiSettingsClient` based on the request and retrieve the properties it needs from that. And on top of that you would have a (new) service as an intermediate for communicating between your plugin and rendering services. Something along the lines of:
`uiSettingsClient` -> `plugin` -> `themingService` -> `renderingService`

As @pgayvallet and myself pointed out, `custom_branding_service` is a good reference point for what you are trying to do, if you stick with the ui-settings-client based approach.

Reply:
Or if you ditch the client, as Pierre is suggesting, you can just have the service calculate this value

Reply:
@majagrubic @pgayvallet 

I think I understand what you're getting at now 😅 

It does seem more straightforward to make a new service (follow what was done for Custom Branding) and I'm happy to switch my approach.

If I do go that way, it would just be reading User Settings in from the UserProfile; ignoring future potential SO user settings and removing the UiSettingsUserClient.

or 

I could double down on the client. I could implement `userProvided()` and `set()`. I would pull the client into the Security Plugin and have the UserProfile use the client for updating the UserProfile data.userSettings. This does seem more complicated.

Please let me know which you would prefer!